kolibri/plugins/coach/assets/src/modules/questionList/actions.js

```diff
@@ -35,11 +36,20 @@ export function setItemStats(store, { classId, exerciseId, quizId, lessonId, gro
   return Promise.all(promises).then(([item, stats]) => {
     if (quizId) {
       store.commit('SET_STATE', { exam: item });
-      stats = stats.map(stat => ({
-        ...stat,
-        exercise_id: stat.content_id,
-        question_id: stat.item,
-      }));
+      // If no one attempted one of the questions, it could get missed out of the list
+      // of difficult questions, so use the exam data to fill in the blanks here.
+      stats = item.question_sources.map(source => {
+        const stat = stats.find(
+          stat => stat.item === source.question_id && stat.content_id === source.exercise_id
+        ) || {
+          correct: 0,
+          total: (stats[0] || {}).total || 0,
```

Comment:
do we need to check `stats.length`?

Reply:
This is essentially doing that - if `stats[0]` is undefined, then either the Array is malformed, or it is empty.

Reply:
ah I see